edit-post/components/plugin-more-menu-item/index.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { compose } from '@wordpress/element';
+import { Slot, Fill, withContext, MenuItemsGroup } from '@wordpress/components';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import MoreMenuItemLayout from './more-menu-item-layout';
+
+/**
+ * Name of slot in which the sidebar should fill.
+ *
+ * @type {String}
+ */
+const SLOT_NAME = 'PluginMoreMenuItem';
+
+function PluginMoreMenuItem( { title, onClick, icon, isActive } ) {
+	return (
+		<Fill name={ SLOT_NAME }>
+			{ ( props ) => {
+				return (
+					<MoreMenuItemLayout
+						isActive={ isActive }
+						title={ title }
+						onClick={ () => {
+							onClick();
+							props.onClose();
+						} }
+						icon={ icon } />
+				);
+			} }
+		</Fill>
+	);
+}
+
+PluginMoreMenuItem = compose( [
+	withContext( 'pluginName' )( ( pluginName, { target } ) => {
+		return {
+			target: `${ pluginName }/${ target }`,
+		};
+	} ),
+	withSelect( ( select, { target } ) => ( {
+		isActive: select( 'core/edit-post' ).getActiveGeneralSidebarName() === target,
```

Comment:
A similar concern with the `type` in here. If it is not `sidebar` why should we even care about the value of `isActive`?

Reply:
It might make more sense to have a general purpose component, but also more targeted for the sidebar and scree take-over so all the logic for `onClick` handler and the current state could be applied independently.